internal/contractgateway/smartcontractgw.go

```diff
@@ -323,7 +322,7 @@ func (g *smartContractGW) storeDeployableABI(msg *messages.DeployContract, compi
 
 func (g *smartContractGW) gatewayErrReply(res http.ResponseWriter, req *http.Request, err error, status int) {
 	log.Errorf("<-- %s %s [%d]: %s", req.Method, req.URL, status, err)
-	reply, _ := json.Marshal(&restErrMsg{Message: err.Error()})
+	reply, _ := json.Marshal(errors.ToRESTError(err))
```

Comment:
Isn't `err` of type `error`? If so, it appears that `ToRESTError()` will return an `errorCode=""`

Reply:
Here's the function. The intention of this (validated in some UTs), is that if `err` conforms to the interface `EthconnectError` then it will set the `Code` on the `RESTError` reply.

```go
func ToRESTError(err error) *RESTError {
	var errorMessage string
	var errorCode = ""
	switch err := err.(type) {
	case EthconnectError:
		errorMessage = err.ErrorNoCode()
		errorCode = err.Code()
	default:
		errorMessage = err.Error()
	}
	return &RESTError{Message: errorMessage, Code: errorCode}
}
```

Reply:
Popping in a new UT to show the e2e

Reply:
https://github.com/hyperledger/firefly-ethconnect/pull/178/commits/3ae5db56aab91f28519f11b27c2ebfc408dc4339

Reply:
Going to also add a UT to the `errors` package as there isn't one there

Reply:
Added UTs to errors package: https://github.com/hyperledger/firefly-ethconnect/pull/178/commits/9c33c837a46de37a734cc2fc37236d79afcb30aa